packages/contract-utils/upgradeable/src/upgradeable.rs

```diff
@@ -0,0 +1,112 @@
+use soroban_sdk::{contractclient, contracterror, Address, BytesN, Env, FromVal, Val};
```

Comment:
should we emit an event for `upgrade` and `downgrade`? 
It may be useful to have this info available on block explorers, wdyt?